packages/angular_devkit/schematics_cli/bin/schematics.ts

```diff
@@ -161,7 +179,14 @@ if (argv['list-schematics']) {
 
 
 /** Create the schematic from the collection. */
-const schematic = collection.createSchematic(schematicName);
+let schematic: Schematic<FileSystemCollectionDescription, FileSystemSchematicDescription>;
+try {
+  schematic = collection.createSchematic(schematicName);
+} catch (e) {
+  logger.fatal(e.message);
```

Comment:
We expose all exceptions in Schematics as separate classes. You could try `instanceof` to display a proper error message, instead of calling `resolve()` yourself which is prone to errors.